presto-product-tests/src/main/java/io/prestosql/tests/hive/TestSyncPartitionMetadata.java

```diff
@@ -92,25 +93,57 @@ public void testInvalidSyncMode()
         cleanup(tableName);
     }
 
+    @Test(groups = {HIVE_PARTITIONING, SMOKE})
+    public void testMixedCasePartitionNames()
+    {
+        String tableName = "test_sync_partition_mixed_case";
+        prepare(hdfsClient, hdfsDataSourceWriter, tableName);
+        String tableLocation = WAREHOUSE_DIRECTORY_PATH + tableName;
+        HiveDataSource dataSource = createResourceDataSource(tableName, "io/prestosql/tests/hive/data/single_int_column/data.orc");
+        hdfsDataSourceWriter.ensureDataOnHdfs(tableLocation + "/col_x=h/col_Y=11", dataSource);
```

Comment:
add a case with `COL_X=UPPER_CASE` that woud verify the _value_ is registered as `UPPER_CASE` and not `upper_case`